chainer/utils/sparse.py

```diff
@@ -123,6 +151,29 @@ def to_coo(x, ldnz=None, requires_grad=False):
             row[i] = coo.row
             col[i] = coo.col
         shape = x.shape[1:]
-        return CooMatrix(data, row, col, shape, requires_grad)
+        return CooMatrix(data, row, col, shape, order=order,
+                         requires_grad=requires_grad)
     else:
         raise ValueError('ndim of x must be 2 or 3.')
+
+
+def _is_c_order(row, col):
```

Comment:
I'm wondering if we should make this function public (and test it) and make it return the order instead, i.e. either `'C'` , `'F'` or `None`. User may call it outside once and reuse the order when constructing additional CooMatrices?

Reply:
Not sure if it's going to be repeatedly constructed with same `row`, `col` though. What do you think? In any case, I'd personally like to test this function.

Reply:
Thank you for your comment. Actually, I though that `'auto'` did not align well with others like `'C'` or `'F'`, because it did not show anything about order. So, I think it is good idea to make the function public and let users call it. And probably it is better to change the name? How about `get_order`?

Reply:
Re: cost of computing the order, the compute complexity is O(nnz) as I mentioned before (nnz is number of non-zero elements), and I think the cost is smaller than the gain in most cases. I though `'auto'` was not your preference, but If you are OK with it, I will make it the default.

Reply:
Yes, I first thought adding unintended overhead was not ideal but if the gain is in general larger then that might be better. Could you proceed with the change?

Reply:
An idea that came up was making `None` behave like `'auto'` and instead introducing a new value such as `'other'`.

Reply:
I think it is good idea to use `'other'`. OK, I will change the code in that way.

Reply:
Got it, thanks a lot.